packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/batch-delete-builds.ts

```diff
@@ -0,0 +1,56 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct } from 'constructs';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for CodeBuildBatchDeleteBuilds
+ */
+export interface CodeBuildBatchDeleteBuildsProps extends sfn.TaskStateBaseProps {
+  /**
+   * The IDs of the builds to delete.
+   */
+  readonly ids: string[];
+}
+
+/**
+ * Delete one or more builds as a task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-codebuild.html
+ */
+export class CodeBuildBatchDeleteBuilds extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: CodeBuildBatchDeleteBuildsProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, CodeBuildBatchDeleteBuilds.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: ['*'],
```

Comment:
deleting builds is usually scoped to a project as per the IAM policies for integrated services [documentation](https://docs.aws.amazon.com/step-functions/latest/dg/codebuild-iam.html#codebuild-iam-batchdeletebuilds). 

Perhaps a `codebuild.IProject` also needs to be an input to this API  similar to `CodeBuildStartBuild`. Ideally we don't want to stamp out a policy that allows deletion of any build from any project.

Reply:
Both BatchDeleteBuilds and StopBuild takes in a build arn as its input, for example ```arn:aws:codebuild:<region>:<AccountID>:build/<ProjectName>:425960b5-af72-4d0e-8229-5e291ff6b0e2``` We have 3 options here
* Option 1, we change the input to require both IProject[] and a list of buildArns, this option will allow us to make a tightly scoped policy without any wildcards, however user needs to make sure the buildId list and project list are mutually inclusive
* Option 2, maintain the current request input. We treat both static and dynamic input equally with ```"arn:aws:codebuild:[[region]]:*:project/*"```, This is the policy we are currently using for dynamic code build policy template https://docs.aws.amazon.com/step-functions/latest/dg/codebuild-iam.html#codebuild-iam-batchdeletebuilds
* Option 3, maintain the current request input. Implement a policy analyzer, base on the input first determine if the ids are passed in dynamically or statically, if it's passed in dynamically we use arn:aws:codebuild:[[region]]:*:project/* otherwise we analyze each input ids, parse the build arn and generate the policy.

I personally prefers Option 1 for these 2 APIs

Reply:
Hey @NovakGu ,

apologies for butting in here, but one quick question :).

Since the ARN of the build is in the form `arn:aws:codebuild:<region>:<AccountID>:build/<ProjectName>:425960b5-af72-4d0e-8229-5e291ff6b0e2` already, can't we just take out the `<ProjectName>` part from it, format an ARN of the Project using its name, and use that in the resource policy?

Reply:
Updated in the new commit.

Essentially for batch-get-reports, batch-delete-builds, and stop-build. When customer put buildIds/reportArns directly into the parameter field. We will parse and buildArns/reportArns to form properly scoped projectArns/reportArns in the generated policy resource.

If customer decides to use path. We will use a scoped wildcard resource policy such as 'arn:aws:codebuild:us-east-1:123456789012:project/*', because there's no way we will know these info at compile time. Values in path are passed at runtime.